### ISSUE TITLE
Test error

### ISSUE THREAD

**Issue by GavinZhou2014** [wkentaro/fcn#48]
hi, i successfully trained my own datasets with fcn32s model, my data has only **two class(background+another one)** and  i **used only 140 images to train** my own model, i just want to see the result first.
But with the test image, i got the answer: **backround ~= 100%**, all is background. I don't know why, maybe a small dataset or other something.
I have some questions:

1. How to distinguish the category of your training image area?  I mean in segmentclass image, how to recognize the area with red color is a cat not a pig

2. Because i have just two class, so what is the impact of small data sets on the final outcome?  very low accuracy? 

Thank you very much!

**Comment by wkentaro** [comment]
> How to distinguish the category of your training image area? I mean in segmentclass image, how to recognize the area with red color is a cat not a pig

Please see the examples and pascal dataset.
https://github.com/wkentaro/fcn/tree/master/examples
https://github.com/wkentaro/fcn/blob/master/fcn/pascal.py

> Because i have just two class, so what is the impact of small data sets on the final outcome? very low accuracy?

I don't think 140 images to train is not so small. The reasons I'm considering are:

1. you are doing something wrong (see examples to fix), in specifically the inputs for the network
2. the dataset is really bad
3. you need to tune the parameters of optimizer like learning rate

**Comment by GavinZhou2014** [comment]
I find the problem,  each color represents a particular kind of training class. i misunderstand it; But **what is the color of the edge between the two classes, is it a specific color?**, i can't find any clue on the code. Thank you very much!

**Comment by GavinZhou2014** [comment]
I re-make my dataset, but when training, i got wrong message "**loss value is nan**", here is the problem: 
![error](https://cloud.githubusercontent.com/assets/9864293/17315575/29f27778-589f-11e6-8d9d-6e79702acd14.png)

It seems that my dataset is ok, but what is the cause of that error, any suggestion? Thanks

**Comment by wkentaro** [comment]
Maybe you need to tune the learning rate.

**Comment by GavinZhou2014** [comment]
I have tuned, but it doesn't work! Under what circumstances the loss value is nan? Thanks

**Comment by wkentaro** [comment]
>  Under what circumstances the loss value is nan? 

When the learning rate is too large. I don't know about other cases.

**Comment by GavinZhou2014** [comment]
Solved! Thanks
